mitodl/micromasters
# Add create channel ui for staff

#### What are the relevant tickets?
Fixes #3473 and #3474 

#### What's this PR do?
Adds a create channel button and dialog to the search page for staff.

#### How should this be manually tested?
- Go to the /learners/ page.
- Verify as a learner you see no button to create a channel, but you do as staff.
- Create a new channel, test validation.
- Ensure you see a confirmation toast and a new chrome window/tab opens with to the new channel in OD.
- Check to be sure your channel got created (e.g. `reddit.local/r/name` should show you the description you entered)

NOTE: your staff user currently won't get added as a contributor on the new channel, so you should expect to see an error page for the channel on the OD side. This is unrelated to this PR.

#### Screenshots (if appropriate)
See #3473
